runtime/lua/vim/highlight.lua

```diff
@@ -76,8 +76,36 @@ function M.range(bufnr, ns, higroup, start, finish, opts)
   end
 end
 
+local function scoped_range(bufnr, ns, higroup, start, finish, opts)
```

Comment:
Maybe it's better to adding scope support to `vim.highlight.range()` instead of duplicating the code?